resources/widget_list.ts

```diff
@@ -1,17 +1,40 @@
 // NOTE
 // This class should be considered deprecated, and any users of this class should
 // just switch over to using CSS grid.
-
-class WidgetList extends HTMLElement {
-  static get observedAttributes() {
+type Sorter = () => number;
+
+const getRandomInt = (max: number) => Math.floor(Math.random() * Math.floor(max));
+
+export default class WidgetList extends HTMLElement {
+  private _nextId = 1;
+  private _nameToId: { [key: string]: number } = {};
+  private _elements: { [key: number]: Sorter } = {};
+  private _sorted: number[] = [];
+  private _elementwidth = 100;
+  private _elementheight = 100;
+  private _xinc1 = 1;
+  private _xinc2 = 0;
+  private _yinc1 = 0;
+  private _yinc2 = 1;
+  private _rowcolsize = 5;
+  private _maxnumber = 1000;
+  private _connected = false;
+  private rootElement: HTMLElement;
+  // bypass shadowRoot maybe null in base class
+  private shadow: ShadowRoot;
```

Comment:
Would you mind explaining more why you can't use `this.shadowRoot`? What would it look like if you did?

Reply:
`this.shadowRoot` is `ShadowRoot|null` and requires additional null checker when using it.

Reply:
Ah, thanks.  I think there is only one other call for `this.shadow.`, so I think it'd be cleaner to turn that into `this.shadowRoot.?` and not have the extra member variable to think about here.

Reply:
Yes, that's reasonable.

Reply:
Would you mind doing that, and then we can merge this?

Reply:
busy in new game version these days.

Reply:
No worries! I just wanted to make sure you weren't waiting on me.  :)